misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JBallerinaDebugServer.java

```diff
@@ -526,13 +539,54 @@ private StackFrame toDapStackFrame(StackFrameProxyImpl stackFrame) {
             dapStackFrame.setSource(source);
             dapStackFrame.setLine((long) stackFrame.location().lineNumber());
             dapStackFrame.setColumn(0L);
-            dapStackFrame.setName(stackFrame.location().method().name());
+            dapStackFrame.setName(getStackFrameName(stackFrame));
             return dapStackFrame;
         } catch (AbsentInformationException | JdiProxyException e) {
             return null;
         }
     }
 
+    /**
+     * Can be used to get the stack frame name.
+     *
+     * @param stackFrame stackFrame.
+     * @return Stack frame name.
+     */
+    private String getStackFrameName(StackFrameProxyImpl stackFrame) {
+        ObjectReference strand;
+        String stackFrameName;
+        try {
+            if (stackFrame.location().method().name().matches(WORKER_LAMBDA_REGEX)
+                    && stackFrame.visibleVariableByName(STRAND_VAR_NAME) == null) {
+                strand = (ObjectReference) ((ArrayReference) stackFrame.getStackFrame().getArgumentValues().get(0))
+                        .getValue(0);
+                stackFrameName = String.valueOf(strand.getValue(strand.referenceType().fieldByName(NAME)));
+                stackFrameName = removeRedundantQuotes(stackFrameName);
+                return WORKER + FRAME_SEPARATOR + stackFrameName;
+            } else if (stackFrame.location().method().name().contains(LAMBDA)
+                    && stackFrame.visibleVariableByName(STRAND_VAR_NAME) == null) {
+                strand = (ObjectReference) ((ArrayReference) stackFrame.getStackFrame().getArgumentValues().get(0))
+                        .getValue(0);
+                stackFrameName = String.valueOf(strand.getValue(strand.referenceType().fieldByName(NAME)));
+                stackFrameName = removeRedundantQuotes(stackFrameName);
+                if (stackFrameName.equals(NULL)) {
+                    stackFrameName = ANONYMOUS;
+                }
+                return START + FRAME_SEPARATOR + stackFrameName;
```

Comment:
Shall we do a simple null check instead of using this String constant `NULL`?

Reply:
Done